packages/generate_package.sh

```diff
@@ -0,0 +1,267 @@
+#!/bin/bash
+
+# Wazuh package generator
+# Copyright (C) 2015, Wazuh Inc.
+#
+# This program is a free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+set -x
+CURRENT_PATH="$( cd $(dirname $0) ; pwd -P )"
+WAZUH_PATH="$(cd $CURRENT_PATH/..; pwd -P)"
+ARCHITECTURE="amd64"
+SYSTEM="deb"
+OUTDIR="${CURRENT_PATH}/output/"
+BRANCH=""
+REVISION="0"
+TARGET="agent"
+JOBS="2"
+DEBUG="no"
+SRC="no"
+BUILD_DOCKER="yes"
+DOCKER_TAG="latest"
+INSTALLATION_PATH="/var/ossec"
+CHECKSUM="no"
+FUTURE="no"
+LEGACY="no"
+IS_STAGE="no"
+
+
+trap ctrl_c INT
+
+clean() {
+    exit_code=$1
+
+    # Clean the files
+    find "${DOCKERFILE_PATH}" \( -name '*.sh' -o -name '*.tar.gz' -o -name 'wazuh-*' \) ! -name 'docker_builder.sh' -exec rm -rf {} +
+
+    exit ${exit_code}
+}
+
+ctrl_c() {
+    clean 1
+}
+
+download_file() {
+    URL=$1
+    DESTDIR=$2
+    if command -v curl > /dev/null 2>&1 ; then
+        (cd ${DESTDIR} && curl -sO ${URL})
+    elif command -v wget > /dev/null 2>&1 ; then
+        wget ${URL} -P ${DESTDIR} -q
+    fi
+}
+
+build_pkg() {
+    if [ "$LEGACY" = "yes" ]; then
+        REVISION="${REVISION}.el5"
+        TAR_URL="https://packages-dev.wazuh.com/utils/centos-5-i386-build/centos-5-i386.tar.gz"
+        TAR_FILE="${CURRENT_PATH}/${SYSTEM}s/${ARCHITECTURE}/legacy/centos-5-i386.tar.gz"
+        if [ ! -f "$TAR_FILE" ]; then
+            download_file ${TAR_URL} "${CURRENT_PATH}/${SYSTEM}s/${ARCHITECTURE}/legacy"
+        fi
+        DOCKERFILE_PATH="${CURRENT_PATH}/${SYSTEM}s/${ARCHITECTURE}/legacy"
+        # TODO: Remove the "pkg" appended to the container name.
+        # This was necessary because we don't have permission to overwrite an existing container.
+        # Also, ensure that the same adjustment is made in the else condition.
```

Comment:
Remove this TO DO, as it is not a problem.

Reply:
done